internal/apiresourceset/tektonapiresourceset.go

```diff
@@ -0,0 +1,418 @@
+	giturls "github.com/whilp/git-urls"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	intstr "k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	roleKind                  = "Role"
+	roleBindingKind           = "RoleBinding"
+	pipelineKind              = "Pipeline"
+	pipelineRunKind           = "PipelineRun"
+	ingressKind               = "Ingress"
+	eventListenerKind         = "EventListener"
+	triggerTemplateKind       = "TriggerTemplate"
+	gitRepoURLPlaceholder     = "<TODO: insert git repo url>"
+	gitDomainPlaceholder      = "<TODO: insert git repo domain>"
+	contextPathPlaceholder    = "<TODO: insert path to the directory containing Dockerfile>"
+	dockerfilePathPlaceholder = "<TODO: insert path to the Dockerfile>"
+	defaultGitRepoBranch      = "master"
+
+	workspaceName = "shared-data"
+)
+
+// TektonAPIResourceSet is the set of CI/CD resources we generate.
+type TektonAPIResourceSet struct {
+}
+
+// CreateResources creates CI/CD resources form the IR.
+func (*TektonAPIResourceSet) CreateResources(ir irtypes.IR) []runtime.Object {
+	tektonTriggersServiceAccountName := "tekton-triggers-admin-service-account"
+	triggerBindingName := "git-event-trigger-binding"
+	triggerTemplateName := "clone-build-push-pipeline-run-trigger-template"
+	pipelineName := "clone-build-push-pipeline"
+	clonePushServiceAccountName := "clone-push-service-account"
+	registrySecretName := "registry-secret"
+	gitEventListenerName := "git-event-listener"
+
+	gitSecrets := createGitSecrets(ir)
+
+	role, serviceAccount, roleBinding := createTektonTriggersRBAC("tekton-triggers-admin-role", tektonTriggersServiceAccountName, "tekton-triggers-admin-role-binding")
+	objs := []runtime.Object{
+		role, serviceAccount, roleBinding,
+		createCloneBuildPushPipeline(pipelineName, ir),
+		createClonePushServiceAccount("clone-push-service-account", gitSecrets, registrySecretName),
+		createGitEventIngress("git-event-ingress", gitEventListenerName),
+		createGitEventTriggerBinding(triggerBindingName),
+		createRegistrySecret(registrySecretName),
+		createGitEventListener(gitEventListenerName, tektonTriggersServiceAccountName, triggerBindingName, triggerTemplateName),
+		createTriggerTemplate(triggerTemplateName, pipelineName, clonePushServiceAccountName, ir),
+	}
+	for _, gitSecret := range gitSecrets {
+		objs = append(objs, gitSecret)
+	}
+	return objs
+}
+
+func createGitSecrets(ir irtypes.IR) [](*corev1.Secret) {
+	secrets := [](*corev1.Secret){}
+	gitDomains := []string{}
+	for _, container := range ir.Containers {
+		gitRepoURL, err := giturls.Parse(container.CICDInfo.GitRepoURL)
+		if err != nil {
+			log.Warnf("Failed to parse git repo url %q Error: %q", container.CICDInfo.GitRepoURL, err)
+			continue
+		}
+		gitDomains = append(gitDomains, gitRepoURL.Hostname())
```

Comment:
don't understand this

Reply:
If we are not able to detect the git hostname , we have to check here and give warning (and separate out) here itself. Otherwise there could be multiple empty values each meant to point to a different repository, and we would end up merging all of them into a single domain, when each were supposed to be aeparate. Entries

Reply:
@HarikrishnanBalagopal I guess this also needs fixing before we merge.